pkg/kubelet/apis/config/v1beta1/defaults.go

```diff
@@ -56,12 +62,18 @@ func SetDefaults_KubeletConfiguration(obj *kubeletconfigv1beta1.KubeletConfigura
 	if obj.HTTPCheckFrequency == zeroDuration {
 		obj.HTTPCheckFrequency = metav1.Duration{Duration: 20 * time.Second}
 	}
+	if obj.EnableServer == nil {
+		obj.EnableServer = utilpointer.BoolPtr(true)
+	}
 	if obj.Address == "" {
 		obj.Address = "0.0.0.0"
 	}
 	if obj.Port == 0 {
 		obj.Port = ports.KubeletPort
 	}
+	if obj.CertDir == "" {
+		obj.CertDir = DefaultCertDir
```

Comment:
@derekwaynecarr, I vaguely remember a discussion about wanting CertDir and VolumePluginDir to default to subdirectories of RootDir when we moved to config (and whether we even wanted them individually controllable in the config file, or just RootDir)

Reply:
Hmm. What other components depend on those paths being stable? Anything that installs volume plugins is one category, I'm less familiar with things that have dependencies on the cert subsystem.

Reply:
Also given that TLSCertFile and TLSPrivateKeyFile are already in the config, would it be weird to hide CertDir?

Reply:
> What other components depend on those paths being stable?

They're already configurable, so I'm not sure what you mean by stable. I guess we can't change behavior for someone today who is starting with a combination of flags and config file and relying on the current defaults for those fields.

I do think it is more coherent to locate those under RootDir by default, but that would have to be done on a new kubelet config version boundary.

> Also given that TLSCertFile and TLSPrivateKeyFile are already in the config, would it be weird to hide CertDir?

I don't think so. TLSCertFile/TLSPrivateKeyFile are for certs you provide. CertDir is for certs the kubelet creates/manages.